src/ui/browserAction/popupPage.js

```diff
@@ -0,0 +1,294 @@
+              .countryFlag=${this.vpnState?.exitServerCountry.code}
+            ></vpn-card>
+            ${this.locationSettings()}
+          </main>
+        </section>
+      </stack-view>
+    `;
+  }
+
+  locationSettings() {
+    if (!BrowserActionPopup.canShowLocationSettings(this.pageURL)) {
+      return null;
+    }
+    const url = new URL(this.pageURL);
+    return BrowserActionPopup.sitePrefrencesTemplate(
+      this.openServerList.bind(this),
+      url.hostname
+    );
+  }
+
+  openServerList() {
+    const onSelectServerResult = (event) => {
+      this.stackView.value?.pop();
+      const city = event?.detail?.city;
+      if (city) {
+        // TODO: Set the City
+      }
+    };
+    const serverListElement = BrowserActionPopup.createServerList(
+      this.vpnState.servers,
+      onSelectServerResult
+    );
+    this.stackView.value?.push(serverListElement);
+  }
+
+  static sitePrefrencesTemplate(openServerList = () => {}, origin = "") {
+    return html`
+      <h1>Preferences for this site</h1>
+      <div class="row">
+        <input type="checkbox" />
+        <p>Always turn off VPN protection for <b>${origin}</b></p>
+      </div>
+      <h2>Select site location</h2>
+      <div class="row" id="selectPageLocation" @click=${openServerList}>
+        <img src="../../assets//flags/CA.png" height="24" width="24" />
+        <p>Toronto</p>
+        <img
+          src="../../assets/img/arrow-icon-right.svg"
+          height="12"
+          width="12"
+          class="arrow"
+        />
+      </div>
+      <button id="selectLocation">Reset Site Prefrences</button>
+    `;
+  }
+
+  /**
+   * Returns true if the Current URL can have a Site-Specific Location
+   * @param {*} pageURL - The Current website
+   * @returns
+   */
+  static canShowLocationSettings(pageURL = "about:blank") {
+    if (!URL.canParse(pageURL)) {
+      return false;
+    }
+    const url = new URL(pageURL);
+    // No http/s nothing to show influence.
+    // i.e data url's or about://
+    if (!(url.protocol === "https:" || url.protocol === "http:")) {
+      return false;
+    }
+    // TODO: Check if we have more restrictions,
+    // i remember there was a restricted url list, somewhere of pages
+    // we cannot interact with i.e accounts.mozilla.org
+    return true;
+  }
+
+  /**
+   *
+   * @param { import("./../../components/serverlist.js").ServerCountryList } list - The ServerList to Render
+   * @param {($0: Event)=>{} } onResult - A callback fired when a city or "back" is clicked.
+   *                                    - Contains an HTML Event with Optinal
+   *                                    - { detail.city : ServerCity }
+   * @returns {HTMLElement} - An already rendered Element, ready to put into the DOM
+   */
+  static createServerList(list = [], onResult = () => {}) {
+    const viewElement = document.createElement("section");
+    render(
+      html`
+        <vpn-titlebar title="Select Location">
```

Comment:
How hard would it be to have a single, fixed, titlebar and update the values dynamically?

Reply:
Not that much. Added that.